### PR TITLE
support validate's report option

### LINKED ISSUE #158: --report option of punx validate is broken?
```
punx validate --report ERROR punx/data/1998spheres.h5
```
The above command should restrict the report to only showing ERROR status findings? Instead it shows all findings.

Could use:
```
punx validate punx/data/1998spheres.h5 | grep ERROR
```
but then why is there a --report option?

### PR DESCRIPTION
**Pull request by prjemian**
- FIX #158

### CODE REVIEW

**Comment by prjemian** [comment]
Curious, one workflow (publish to PyPI) is failing early on.  Restarted it and it failed again while others succeed.  Maybe it is a temporary server problem.  Come back later.

**Comment by prjemian** [comment]
CI [failure](https://github.com/prjemian/punx/runs/4588745037?check_suite_focus=true) occurs when constructing docker container for publishing distribution to PyPI.  Not related to this PR particularly.

**Comment by prjemian** [comment]
@carterbox - Ready for review.

File: docs/source/validate.rst [hunk 1]
```diff
@@ -34,26 +34,19 @@ validate a NeXus file
...
-                           COMMENT,ERROR,NOTE,OK,TODO,UNUSED,WARN
-     -l [LOGFILE], --logfile [LOGFILE]
-                           log output to file (default: no log file)
-     -i INTEREST, --interest INTEREST
-                           logging interest level (1 - 50), default=1 (Level 1)
+     -h, --help       show this help message and exit
+    --report REPORT  select which validation findings to report, choices: COMMENT,ERROR,NOTE,OK,OPTIONAL,TODO,UNUSED,WARN (separate with comma if more
+                     than one, do not use white space)
 
 The **REPORT** findings are as presented in the table above for each validation step.
```
Comment (carterbox) [inline docs/source/validate.rst:49]: Seems kinda annoying that you have to maintain the help text in two places. No suggestion; just an observation.

File: punx/tests/test_validate.py [hunk 2]
```diff
@@ -654,4 +654,54 @@ def TODO_test_default_plot_v1_fail(hfile):
...
+        ["writer_1_3.hdf5", "NOTE", 1],
+        ["writer_1_3.hdf5", "NOTE,TODO", 7 + 1],
+        ["writer_2_1.hdf5", "note", 0],
+        ["writer_2_1.hdf5", "TODO", 11],
+        ["1998spheres.h5", "ERROR", 2],
+        ["02_03_setup.h5", "NOTE,OPTIONAL,ERROR", 98 + 70 + 0],
+        ["prj_test.nexus.hdf5", "", 121],
+    ],
+)
+def test_report_option(infile, report, observations, capsys):
```
Comment (carterbox) [inline punx/tests/test_validate.py:670]: What value does `capsys` get if pytest only passes three arguments?

**Review by carterbox — APPROVED** [review]

**Comment by carterbox** [comment]
I ran the tests locally and tested the `--report` CLI option with a personal h5 file. Looks good to me! Thanks!

File: docs/source/validate.rst (hunk 1 above)
Comment (prjemian) [reply docs/source/validate.rst:49]: agreed

File: punx/tests/test_validate.py (hunk 2 above)
Comment (prjemian) [reply punx/tests/test_validate.py:670]: It's a pytest [fixture](https://docs.pytest.org/en/6.2.x/capture.html#accessing-captured-output-from-a-test-function).  It supplies the pytest feature that captures output to stdout & stderr.

**Comment by prjemian** [comment]
@carterbox Thanks for the review and the improvement.

**Pull request merged**